src/arrangement/core.cljc

```diff
@@ -78,10 +90,11 @@
         (compare-seqs rank a b)
 
         :else
-        (let [class-diff (compare (.getName (class a))
-                                  (.getName (class b)))]
+        (let [class-diff (compare (type-name a)
+                                  (type-name b))]
           (if (zero? class-diff)
-            (if (instance? java.lang.Comparable a)
-              (compare a b)
-              (compare (str a) (str b)))
```

Comment:
I ran into an issue with one of the tests failing randomly because the printed representation of `java.util.Currency` contains it's system id, so sorting was non deterministic. I'm not sure what the best approach is here, since if we have two instances of the same type which don't implement `[I]Comparable`, there's not a great way to sort them.

In the above example I changed this to `str` since comparing `USD` to `JPY` is meaningful in currencies. I could change back to `pr-str`, but then we'd have to change the test to something else.